packages/patternfly-4/react-core/src/components/BackgroundImage/BackgroundImage.tsx

```diff
@@ -13,87 +13,86 @@ import {
   c_background_image_BackgroundImage_lg
 } from '@patternfly/react-tokens';
 
-export const BackgroundImageSrc = {
-  xs: 'xs',
-  xs2x: 'xs2x',
-  sm: 'sm',
-  sm2x: 'sm2x',
-  lg: 'lg',
-  filter: 'filter'
+export enum BackgroundImageSrc {
+  xs = 'xs',
+  xs2x = 'xs2x',
+  sm = 'sm',
+  sm2x = 'sm2x',
+  lg = 'lg',
+  filter = 'filter'
 };
 
-const variableMap = {
+const cssVariables = {
   [BackgroundImageSrc.xs]: c_background_image_BackgroundImage && c_background_image_BackgroundImage.name,
   [BackgroundImageSrc.xs2x]: c_background_image_BackgroundImage_2x && c_background_image_BackgroundImage_2x.name,
   [BackgroundImageSrc.sm]: c_background_image_BackgroundImage_sm && c_background_image_BackgroundImage_sm.name,
   [BackgroundImageSrc.sm2x]: c_background_image_BackgroundImage_sm_2x && c_background_image_BackgroundImage_sm_2x.name,
-  [BackgroundImageSrc.lg]: c_background_image_BackgroundImage_lg && c_background_image_BackgroundImage_lg.name
+  [BackgroundImageSrc.lg]: c_background_image_BackgroundImage_lg && c_background_image_BackgroundImage_lg.name,
 };
 
-export const propTypes = {
-  /** Additional classes added to the background. */
-  className: PropTypes.string,
-  /** Override image styles using a string or BackgroundImageSrc */
-  src: PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.shape({
-      xs: PropTypes.string,
-      xs2x: PropTypes.string,
-      sm: PropTypes.string,
-      sm2x: PropTypes.string,
-      lg: PropTypes.string,
-      filter: PropTypes.string
-    })
-  ]).isRequired
+export interface BackgroundImageSrcMap {
+  xs: string,
+  xs2x: string,
+  sm: string,
+  sm2x: string,
+  lg: string,
+  filter: string
 };
 
-export const defaultProps = {
-  className: ''
+export interface BackgroundImageProps extends Omit<React.HTMLProps<HTMLDivElement>, 'src'> {
+  /** Additional classes added to the background. */
+  className?: string; 
+  /** Override image styles using a string or BackgroundImageSrc */
+  src: string | BackgroundImageSrcMap; 
 };
 
-const BackgroundImage = ({ className, src, ...props }) => {
+export const BackgroundImage: React.FunctionComponent<BackgroundImageProps> = ({
+  className = '', 
+  src,
+  ...props
+}: BackgroundImageProps) => {
+  let srcMap = src;
   // Default string value to handle all sizes
-  const variableOverrides =
-    typeof src === 'string'
-      ? Object.keys(BackgroundImageSrc).reduce(
-          (prev, size) => ({
-            ...prev,
-            [BackgroundImageSrc[size]]: src
-          }),
-          {}
-        )
-      : src;
+  if (typeof src === 'string') {
+    srcMap = {
+      [BackgroundImageSrc.xs]: src,
+      [BackgroundImageSrc.xs2x]: src,
+      [BackgroundImageSrc.sm]: src,
+      [BackgroundImageSrc.sm2x]: src,
+      [BackgroundImageSrc.lg]: src,
+      [BackgroundImageSrc.filter]: '', // unused
+    };
+  }
 
+  // Build stylesheet based on cssVariables
```

Comment:
csssheet What type is this?

Reply:
It's a string. I added that to the comment above it to be more explicit if that works.